### PR TITLE
Change cartographer tile reveal logic to match H3

### PR DESCRIPTION
**Pull request by dydzio0614**
Fixes https://github.com/vcmi/vcmi/issues/1339

### CODE REVIEW

File: lib/Terrain.cpp [hunk 1]
```diff
@@ -440,6 +440,16 @@ bool TerrainType::isUnderground() const
...
 }
 
+bool TerrainType::isSurfaceCartographerCompatible() const
+{
+	return isLand() && id != Terrain::SUBTERRANEAN && id != Terrain::ROCK;
+}
+
+bool TerrainType::isUndergroundCartographerCompatible() const
+{
+	return isLand() && id == Terrain::SUBTERRANEAN;
```
Comment (IvanSavenko) [inline lib/Terrain.cpp:450]: 1) What about ROCK? Not 100% sure but I suspect that it should be "revealed" as well by this cartographer - to replace stars from terra incognita with revealed unpassable rocks

2) Don't use terrain types directly, instead use traits from config - we may have new terrains added by mods.
I see isPassable() check that can replace ROCK and isUnderground() that can replace direct check for SUBTERRANEAN

**Review by kambala-decapitator — CHANGES_REQUESTED** [review]
you have a mix of tabs and spaces for indentation, please change to tabs only

**Comment by dydzio0614** [comment]
ad. 1: ROCK is not revealed in original H3

ad. 2: isUnderground() also includes lava in our config

File: lib/Terrain.cpp (hunk 1 above)
Comment (dydzio0614) [reply lib/Terrain.cpp]: ad. 1: ROCK is not revealed in original H3

ad. 2: isUnderground() also includes lava in our config

Comment (IvanSavenko) [reply lib/Terrain.cpp]: Hmm... Interesting, so there is no way to reveal ROCK in H3?
Okay then.

2 - In that case use isSurface()

**Comment by dydzio0614** [comment]
Everything good now?

**Comment by IvanSavenko** [comment]
Sounds OK for me. 
@kambala-decapitator?

**Review by kambala-decapitator — APPROVED** [review]

**Comment by IvanSavenko** [comment]
Waiting my usual ~24 hours and merging tomorrow if there are no objections/further comments.

**Pull request merged**